### PR TITLE
don't raise when location is missing.

### PR DESCRIPTION
**Pull request by osahyoun**
Handle empty resource URIs and Return 200 to queue for failed page updates.

### CODE REVIEW

**Comment by Tuuleh** [comment]
Actually Omar, after thinking about this a little bit, I'm not totally sure what we should be doing for failed page updates. Wouldn't this lead to scenarios where it'd become possible to have petitions and donations that would not be tied to AK pages without being alerted to it by anything? A 500 from the worker would basically be a last safeguard that there's an invalid page.

**Comment by osahyoun** [comment]
I was thinking the same when implementing this. This happens when a page, for whatever reason, hasn't been updated with an action kit resource URI. I think the page should be flagged on champaign and blocked from being published. 

For the worker, it might be good to log to a db table any non 2xx responses (with associated resource) from AK.

What do you think?

**Comment by Tuuleh** [comment]
I think it'd be good to keep count of all AK-related failures, but I'm not sure I'd put that in the DB. What's your reasoning around that?

File: spec/requests/pages_spec.rb [hunk 1]
```diff
@@ -3,6 +3,34 @@
...
+        before do
+          VCR.use_cassette('page_update_404') do
+            post "/message", params
+          end
+        end
+
+        it 'fails but is 200' do
+          expect(response.body).to match(/Sorry, this request could not be processed/)
+          expect(response.status).to eq(200)
+        end
```
Comment (osahyoun) [inline spec/requests/pages_spec.rb]: @Tuuleh I think then I should return 5xx so the job eventually ends up on the deadletter queue for inspection. And we should stop jobs we know will never run from being posted to the queue in the first place. wdt?

Comment (Tuuleh) [inline spec/requests/pages_spec.rb]: That sounds good to me. :+1:

**Comment by osahyoun** [comment]
@Tuuleh this one is good to go. I'm now raising argument error when a page is missing a valid resource ID, with a more meaningful message. This'll alert us in the logs and will eventually put the message into the dead letter queue for inspection.

File: app/workers/queue_listener.rb [hunk 2]
```diff
@@ -43,11 +43,17 @@ def update_share(params)
...
+    extract_id(params[:petition_uri])
   end
 
   def donation_id(params)
-    params.fetch(:donation_uri, '').match(/(\d+)\/$/)[1]
+    extract_id(params[:donation_uri])
+  end
+
+  def extract_id(uri)
+    id = (uri || '').match(/(\d+)\/$/){|m| m[1] if m }
```
Comment (Tuuleh) [inline app/workers/queue_listener.rb]: Can you explain this line? I'm not totally clear on what you're matching against.

Comment (osahyoun) [inline app/workers/queue_listener.rb]: I'm trying to extract the ID from the URI. The block allows me to check if there was a match before accessing it.

Comment (Tuuleh) [inline app/workers/queue_listener.rb]: That much I was clear on, I was wondering about the regex.

Comment (osahyoun) [inline app/workers/queue_listener.rb]: `/(\d+)\/$/` match one or more numbers, followed by a trailing slash.

**Pull request merged**